### PR TITLE
User model validation

### PR DESCRIPTION
**Pull request by msosic97**

### CODE REVIEW

File: Models/user.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	Password  string `json:"password"`
+	Email     string `json:"email"`
+	FirstName string `json:"firsname"`
+	LastName  string `json:"lastname"`
+	Phone     string `json:"phone"`
+	CreatedAt int64 `json:"created_at"`
+}
+
+
+type Login struct {
```
Comment (amina-hybird) [inline Models/user.go]: name should be something like LoginRequest

File: Tools/DB/db.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+
+// DSN is a string that contains the necessary information to establish a connection to a specific data source or database.
+
+var DB *sql.DB
+
+//  Connection to DB
+func StartDB(config *config.Config) (*sql.DB, error) {
+	dsn := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s",
```
Comment (amina-hybird) [inline Tools/DB/db.go:45]: you can add validations for this, for example func Validate where you will check if there is DBName provided (if len = 0) or something like that

File: mynotes.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+My remider notes
```
Comment (amina-hybird) [inline mynotes.txt:1]: you can put this in .gitignore

File: Models/user.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+// User validation
+func (u *User) Validate() bool {
+	// Validate ID
+	if u.ID <= 0 {
+		// return errors.New("ID must be a positive integer")
+		return false
+	}
+
+	// Validate Username
+	if len(u.Username) < 4 {
```
Comment (amina-hybird) [inline Models/user.go]: you can check also > than some number to have limit for all fields

File: Routes/helper/helpers.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package helper
+
+import (
+	"golang.org/x/crypto/bcrypt"
+)
+
+func HashPassword(password string) (string, error) {
+	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return "", err
```
Comment (amina-hybird) [inline Routes/helper/helpers.go:12]: you can log error here

File: Tools/DB/db.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+
+
+var DB *sql.DB
+
+//  Connection to DB
+func StartDB(config *config.Config) (*sql.DB, error) {
+
+// Validation
+	if len(config.DBName) == 0 {
```
Comment (amina-hybird) [inline Tools/DB/db.go:20]: this can be in separate function Validate and then call it here

Comment (msosic97) [reply Tools/DB/db.go:20]: to write separate Validate function only for DBName or for all other fields?

**Pull request merged**